processor/src/test/java/org/mapstruct/ap/test/nestedbeans/DottedErrorMessageTest.java

```diff
@@ -0,0 +1,186 @@
+import org.mapstruct.ap.test.nestedbeans.erroneous.InfoDto;
+import org.mapstruct.ap.test.nestedbeans.erroneous.UnmappableCollectionElementPropertyMapper;
+import org.mapstruct.ap.test.nestedbeans.erroneous.UnmappableDeepListMapper;
+import org.mapstruct.ap.test.nestedbeans.erroneous.UnmappableDeepMapKeyMapper;
+import org.mapstruct.ap.test.nestedbeans.erroneous.UnmappableDeepMapValueMapper;
+import org.mapstruct.ap.test.nestedbeans.erroneous.UnmappableValuePropertyMapper;
+import org.mapstruct.ap.test.nestedbeans.erroneous.UserDto;
+import org.mapstruct.ap.test.nestedbeans.erroneous.User;
+import org.mapstruct.ap.test.nestedbeans.erroneous.WheelDto;
+import org.mapstruct.ap.test.nestedbeans.erroneous.Wheel;
+import org.mapstruct.ap.test.nestedbeans.erroneous.Car;
+import org.mapstruct.ap.test.nestedbeans.erroneous.CarDto;
+import org.mapstruct.ap.test.nestedbeans.erroneous.House;
+import org.mapstruct.ap.test.nestedbeans.erroneous.HouseDto;
+import org.mapstruct.ap.test.nestedbeans.erroneous.Color;
+import org.mapstruct.ap.test.nestedbeans.erroneous.ColorDto;
+import org.mapstruct.ap.test.nestedbeans.erroneous.Roof;
+import org.mapstruct.ap.test.nestedbeans.erroneous.RoofDto;
+import org.mapstruct.ap.test.nestedbeans.erroneous.UnmappableDeepNestingMapper;
+import org.mapstruct.ap.test.nestedbeans.erroneous.Word;
+import org.mapstruct.ap.test.nestedbeans.erroneous.WordDto;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.compilation.annotation.CompilationResult;
+import org.mapstruct.ap.testutil.compilation.annotation.Diagnostic;
+import org.mapstruct.ap.testutil.compilation.annotation.ExpectedCompilationOutcome;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+@WithClasses({
+    Car.class, CarDto.class, Color.class, ColorDto.class,
+    House.class, HouseDto.class, Roof.class, RoofDto.class,
+    User.class, UserDto.class, Wheel.class, WheelDto.class,
+    Dictionary.class, DictionaryDto.class, Word.class, WordDto.class,
+    ForeignWord.class, ForeignWordDto.class,
+    Computer.class, ComputerDto.class,
+    Cat.class, CatDto.class, Info.class, InfoDto.class
+})
+@RunWith(AnnotationProcessorTestRunner.class)
+public class DottedErrorMessageTest {
+
+    private static final String PROPERTY = "property";
+    private static final String COLLECTION_ELEMENT = "Collection element";
+    private static final String MAP_KEY = "Map key";
+    private static final String MAP_VALUE = "Map value";
+
+    @Test
+    @WithClasses({
+        UnmappableDeepNestingMapper.class
+    })
+    @ExpectedCompilationOutcome(
+        value = CompilationResult.FAILED,
+        diagnostics = {
+            @Diagnostic(type = UnmappableDeepNestingMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 26,
+                messageRegExp = "Can't map " + PROPERTY +
+                    " \".*Color house\\.roof\\.color\" to \".*house\\.roof\\.color\"\\. " +
```

Comment:
Inherited Mapping customizations from a MapperConfig class could have been applied along the way (unsure if that works, yet, but that's on the list of open points).